src/Microsoft.Health.Dicom.Api/Web/LimitStream.cs

```diff
@@ -0,0 +1,114 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.IO;
+using System.Runtime.CompilerServices;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+
+namespace Microsoft.Health.Dicom.Api.Web
+{
+    class LimitStream : Stream
+    {
+        private Stream _stream;
+
+        private long _bytesLeft;
+
+        public override bool CanRead => _stream.CanRead;
+
+        public override bool CanSeek => _stream.CanSeek;
+
+        public override bool CanWrite => _stream.CanWrite;
+
+        public override long Length => _stream.Length;
+
+        public override long Position { get => _stream.Position; set => throw new NotImplementedException(); }
```

Comment:
nit: should be `NotSupportedException`

Reply:
Alternatively, if you want to support `Position` and `Seek`, then we should nail down the semantics to do some math. Does this `Stream` limit the number of `bytes` read? Or does it limit the `Length` of the `Stream`?

Reply:
Implemented as discussed offline